OpenRA.Mods.Common/Widgets/ConfirmationDialogs.cs

```diff
@@ -15,28 +15,60 @@ namespace OpenRA.Mods.Common.Widgets
 {
 	public static class ConfirmationDialogs
 	{
-		public static void PromptConfirmAction(string title, string text, Action onConfirm, Action onCancel = null, string confirmText = null, string cancelText = null)
+		public static void PromptConfirmAction(
+			string title,
+			string text,
+			Action onConfirm,
+			Action onCancel = null,
+			Action onOther = null,
+			string confirmText = null,
+			string cancelText = null,
+			string otherText = null)
 		{
 			var prompt = Ui.OpenWindow("CONFIRM_PROMPT");
+			var confirmButton = prompt.GetOrNull<ButtonWidget>("CONFIRM_BUTTON");
+			var cancelButton = prompt.GetOrNull<ButtonWidget>("CANCEL_BUTTON");
+			var otherButton = prompt.GetOrNull<ButtonWidget>("OTHER_BUTTON");
+
 			prompt.Get<LabelWidget>("PROMPT_TITLE").GetText = () => title;
 			prompt.Get<LabelWidget>("PROMPT_TEXT").GetText = () => text;
 			if (!string.IsNullOrEmpty(confirmText))
-				prompt.Get<ButtonWidget>("CONFIRM_BUTTON").GetText = () => confirmText;
-			if (!string.IsNullOrEmpty(cancelText))
-				prompt.Get<ButtonWidget>("CANCEL_BUTTON").GetText = () => cancelText;
+				confirmButton.GetText = () => confirmText;
+			if (!string.IsNullOrEmpty(otherText) && otherButton != null)
+				otherButton.GetText = () => otherText;
+			if (!string.IsNullOrEmpty(cancelText) && cancelButton != null)
+				cancelButton.GetText = () => cancelText;
 
-			prompt.Get<ButtonWidget>("CONFIRM_BUTTON").OnClick = () =>
+			confirmButton.OnClick = () =>
```

Comment:
You'll now need null checks before using these variables (since `GetOrNull` can return null – `Get` would immediately throw an exception).